### PR TITLE
Offline Geocoder

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
Makes use of µFuzzy for JSON based custom geocoder results

### CODE REVIEW

File: packages/geocoder/src/geocoders/offline.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    };
+  }
+
+  rewriteAutocompleteResponse(response: OfflineResponse): MultiGeocoderResponse {
+    return {
+      features: response?.map((r, index) => ({
+        geometry: { type: "Point", coordinates: [r.lon, r.lat] },
+        properties: {
+          layer: "custom",
+          id: `custom-${index}`,
```
Comment (binh-dam-ibigroup) [inline packages/geocoder/src/geocoders/offline.ts:32]: Sort props

File: packages/geocoder/src/geocoders/types.ts [hunk 2]
```diff
@@ -45,6 +46,7 @@ export type AutocompleteQuery = {
   layers?: string;
   options?: RequestInit;
   size?: number;
+  items?: OfflineResponse
```
Comment (binh-dam-ibigroup) [inline packages/geocoder/src/geocoders/types.ts:49]: Sort props

**Review by binh-dam-ibigroup — APPROVED** [review]
Ok with some cleanup

File: packages/geocoder/src/apis/offline/index.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+type OfflineQuery = {
+  items: OfflineResponse;
+  text?: string;
+};
+/**
+ * Search for an address using offline geocoder
+ *
+ * @param  {Object} $0
+ * @param  {string} $0.text query
+ * @para   {Object} $0.items list of items to search
```
Comment (amy-corson-ibigroup) [inline packages/geocoder/src/apis/offline/index.ts:18]: I think you're missing an "m" here

**Review by amy-corson-ibigroup — APPROVED** [review]
Looks good!

**Pull request merged**